### PR TITLE
Melee + Start of Enh Shaman

### PR DESCRIPTION
**Pull request by lologarithm**

### CODE REVIEW

File: generate_items/main.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ func main() {
 	flag.Parse()
 
 	if *outDir == "" {
-		panic("outDir flag is required!")
+		// Default to items package so I can debug this file. :D
```
Comment (jimmyt857) [inline generate_items/main.go:12]: Remove this

File: sim/core/character.go [hunk 2]
```diff
@@ -165,8 +170,27 @@ func (character *Character) AddStats(stat stats.Stats) {
 	character.stats = character.stats.Add(stat)
 }
 func (character *Character) AddStat(stat stats.Stat, amount float64) {
+	if stat == stats.MeleeHaste {
```
Comment (jimmyt857) [inline sim/core/character.go:173]: I think it should be fine to put the check in here and delete AddMeleeHaste(). This isn't really on the hot path so I don't think the performance difference is that high? Either way I'd rather do it the simple way first, and optimize later.

File: sim/core/test_utils.go [hunk 3]
```diff
@@ -137,7 +137,7 @@ type AllEncountersTestOptions struct {
 
 func IndividualSimAllEncountersTest(testOpts AllEncountersTestOptions) {
 	isr := NewIndividualSimRequest(testOpts.Inputs)
-	// isr.SimOptions.Debug = true
```
Comment (jimmyt857) [inline sim/core/test_utils.go:140]: Remove this

File: sim/shaman/elemental/elemental_test.go [hunk 4]
```diff
@@ -121,7 +121,7 @@ func TestSimulatePreRaidNoBuffs(t *testing.T) {
 		},
 
 		ExpectedDpsShort: 994.9,
-		ExpectedDpsLong:  364.5,
+		ExpectedDpsLong:  418.5,
```
Comment (jimmyt857) [inline sim/shaman/elemental/elemental_test.go:124]: What changed here?

File: sim/shaman/shaman.go [hunk 5]
```diff
@@ -19,6 +19,34 @@ func NewShaman(character core.Character, talents proto.ShamanTalents, selfBuffs
...
 
+	if shaman.Talents.ThunderingStrikes > 0 {
+		shaman.AddStat(stats.MeleeCrit, core.MeleeCritRatingPerCritChance*1*float64(shaman.Talents.ThunderingStrikes))
+	}
+
+	if shaman.Talents.DualWieldSpecialization > 0 {
+		shaman.AddStat(stats.MeleeHit, core.MeleeCritRatingPerCritChance*2*float64(shaman.Talents.DualWieldSpecialization))
+	}
+
+	if shaman.Talents.MentalQuickness > 0 {
```
Comment (jimmyt857) [inline sim/shaman/shaman.go:30]: Mental quickness is also already added (below).

File: ui/enhancement_shaman/sim.ts [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+import * as ShamanInputs from './inputs.js';
+import * as Presets from './presets.js';
+
+export class EnhancementShamanSimUI extends IndividualSimUI<Spec.SpecEnhancementShaman> {
+  constructor(parentElem: HTMLElement, player: Player<Spec.SpecEnhancementShaman>) {
+		super(parentElem, player, {
+			// Can be 'Alpha', 'Beta', or 'Live'. Just adds a postfix to the generated title.
+			releaseStatus: 'Beta',
+			cssClass: 'Enhancement-shaman-sim-ui',
```
Comment (jimmyt857) [inline ui/enhancement_shaman/sim.ts:35]: make this lowercase to match the others

File: ui/enhancement_shaman/_sim.scss [hunk 7]
```diff
@@ -0,0 +1,23 @@
+// Set color variables which control the site theme.
+.elemental-shaman-sim-ui {
```
Comment (jimmyt857) [inline ui/enhancement_shaman/_sim.scss:2]: This should be 'enhancement-shaman-sim-ui' to match the value in sim.ts. This is probably why your colors aren't working.

File: sim/core/character.go (hunk 2 above)
Comment (lologarithm) [reply sim/core/character.go:173]: The function signature has to change so I made a new function

File: ui/enhancement_shaman/sim.ts (hunk 6 above)
Comment (lologarithm) [reply ui/enhancement_shaman/sim.ts:35]: sorry, broad find-replace

File: sim/shaman/elemental/elemental_test.go (hunk 4 above)
Comment (lologarithm) [reply sim/shaman/elemental/elemental_test.go:124]: mana regen!

**Pull request merged**